lingocore/lib/lesson_pages.dart

```diff
@@ -0,0 +1,179 @@
+
+    if (currentQuestion > totalQuestions) {
+      Navigator.push(
+        context,
+        MaterialPageRoute(
+          builder: (context) => LessonEnd(score: score, totalQ: totalQuestions, currentQ: currentQuestion,),
+        ),
+      );
+    }
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    return Scaffold(
+      backgroundColor: backgroundColor,
+      body: Center(
+        child: Column(
+          children: [
+            Padding(
+              padding: const EdgeInsets.symmetric(
+                horizontal: 20.0,
+                vertical: 50,
+              ),
+              child: TweenAnimationBuilder<double>(
+                tween: Tween<double>(
+                  begin: 0.0,
+                  end: (currentQuestion - 1) / totalQuestions,
+                ),
+                duration: const Duration(milliseconds: 150),
+                builder: (context, value, child) {
+                  return LinearProgressIndicator(
+                    value: value,
+                    backgroundColor: const Color.fromARGB(255, 195, 206, 215),
+                    color: Colors.blue,
+                    minHeight: 20,
+                    borderRadius: BorderRadius.circular(10),
+                  );
+                },
+              ),
+            ),
+            QuestionWidget(question: question, checkAns: check),
+          ],
+        ),
+      ),
+    );
+  }
+}
+
+class LessonEnd extends StatelessWidget {
+  final int score, totalQ, currentQ;
+  const LessonEnd({super.key, required this.score, required this.totalQ, required this.currentQ});
+
+  @override
+  Widget build(BuildContext context) {
+    return Scaffold(
+      backgroundColor: Colors.white,
+      body: Center(
+        child: Column(
+          mainAxisAlignment: MainAxisAlignment.start,
+          children: [
+            Padding(
+              padding: const EdgeInsets.fromLTRB(10, 20, 10, 100),
+              child: LinearProgressIndicator(
+                borderRadius: BorderRadius.circular(10),
+                value: currentQ - 1 / totalQ,
+                minHeight: 20,
+                backgroundColor: const Color.fromARGB(255, 195, 206, 215),
+                color: Colors.blue,
+              ),
+            ),
+            Padding(
+              padding: const EdgeInsets.only(bottom: 25),
+              child: Text(
+                "$score / $totalQ",
+                style: TextStyle(fontSize: 20, color: Colors.black),
+              ),
+            ),
+            Padding(
+              padding: const EdgeInsets.fromLTRB(10, 20, 10, 100),
+              child: Text(
+                "Testi bitirdiniz!",
+                style: TextStyle(fontSize: 30, color: Colors.black),
+              ),
+            ),
+            ElevatedButton(
+              onPressed:
+                  () => Navigator.push(
+                    context,
+                    MaterialPageRoute(builder: (context) => LessonPages()),
+                  ),
```

Comment:
Hani nereye geri dönüyor? Niye üst üste sonsuz olarak push yapıyorsun? Ana ekrana dönmek için pop yapsana